docs/design/control-plane-resilience.md

```diff
@@ -0,0 +1,220 @@
+# Kubernetes/Ubernetes Control Plane Resilience
+##  Long Term Design and Current Status
+###by Quinton Hoole, Mike Danese and Justin Santa-Barbara  
+###December 14, 2015
+
+## Summary
+
+Some amount of confusion exists around how we currently, and in future
+want to ensure resilience of the Kubernetes (and by implication
+Ubernetes) control plane.  This document is an attempt to capture that
+definitively. It covers areas including self-healing, high
+availability, bootstrapping and recovery.  Most of the information in
+this document already exists in the form of github comments,
+PR's/proposals, scattered documents, and corridor conversations, so
+document is primarily a consolidation and clarification of existing
+ideas.
+
+## Terms
+
+* **Self-healing:** automatically restarting or replacing failed
+  processes and machines without human intervention
+* **High availability:** continuing to be available and work correctly
+  even if some components are down or uncontactable.  This typically
```

Comment:
I was actually trying to draw an explicit distinction between dead software ('down') and healthy software that's simply not visible due to a network partition ('uncontactable'). @timothysc I think I'll leave it as is, unless you have violent objections.